apps/estuary/lib/estuary/init_server.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Estuary.InitServer do
+  @moduledoc """
+  Task to initialize estuary and start ingesting each previously recorded dataset
+  """
+  use GenServer
+
+  alias Estuary.DataReader
+  alias Estuary.Datasets.DatasetSchema
+  alias Estuary.DataWriter
+
+  def start_link(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    GenServer.start_link(__MODULE__, [], name: name)
+  end
+
+  def init(_) do
+    with :ok <- create_table(),
+         :ok <- DataReader.init(),
+         pid <- Process.whereis(Pipeline.DynamicSupervisor) do
+      Process.monitor(pid)
+      {:ok, %{pipeline: pid}}
+    end
+  end
+
+  def handle_info({:DOWN, _, _, pid, _}, %{pipeline: pid}) do
```

Comment:
I think I'm missing context on why it's necessary to babysit Pipeline in this way and go through the init steps again when it goes down.

Reply:
pipeline will get restarted but the pipeline supervisor is a dynamic supervisor so if it goes down, it will come back up with no memory of the things it was supposed to be running so we need some other process to reschedule our readers/writers to come back up under it.